### PR TITLE
Add Glossary and update Command Summary

### PR DESCRIPTION
**Pull request by RubyNguyen07**

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -464,9 +478,12 @@ If you have a project or client selected, pressing ***Enter*** in **search mode*
 
 # Command summary
```
Comment (immanuelhume) [inline docs/UserGuide.md:480]: How about a short note explaining the syntax?

```suggestion

The table below summarizes the layout of all the commands in Mycelium. You may
read the statement `q -a foo [-b bar]` like so:

* `q` is the command,
* `-a foo` indicates a compulsory argument;
* `[-b bar]` indicates an optional argument.

```

File: docs/UserGuide.md [hunk 2]
```diff
@@ -464,9 +478,12 @@ If you have a project or client selected, pressing ***Enter*** in **search mode*
...
 
-Action | Format
---------|------------------
-**Create new client** | c -cn client_name -e email -y year_of_birth -src source -mn mobile_number <br> e.g., `c -cn 'Alice Baker' -e alice_baker@bakers.com -y 2000`
-**Delete an existing client** | dc -e email <br> e.g., `dc -e alice_baker@bakers.com`
-**Create a project** | p -pn project_name -e client_email -s status -src source -d description -ad accepted_date -dd deadline_date <br> e.g., `p -pn 'Mycelium Desktop' -e spiderman@gmail.com -src fiverr.com -dd 30/02/2075`
-**Delete a project** | dp -pn project_name <br> `e.g., dp -pn Mycelium`
+Action | Format | Example 
+--------|------------------|------
+**Create new client** | c -cn client_name -e email -y year_of_birth -src source -mn mobile_number | `c -cn 'Alice Baker' -e alice_baker@bakers.com -y 2000`
```
Comment (immanuelhume) [inline docs/UserGuide.md:483]:
```suggestion
**Create new client** | `c -cn client_name -e email [-y year_of_birth] [-src source] [-mn mobile_number]` | `c -cn Alice Baker -e alice_baker@bakers.com -y 2000`
```

File: docs/UserGuide.md [hunk 3]
```diff
@@ -464,9 +478,12 @@ If you have a project or client selected, pressing ***Enter*** in **search mode*
...
+**Delete an existing client** | dc -e email | `dc -e alice_baker@bakers.com`
+**Update an existing client** | uc -e email [-cn client_name] [-y year_of_birth] [-src source] [-mn mobile_number] [-e2 new_email] | `uc -e alice_baker@bakers.com -y 2002 -mn 85090234`
+**Create a project** | p -pn project_name -e client_email -s status -src source -d description -ad accepted_date -dd deadline_date | `p -pn 'Mycelium Desktop' -e spiderman@gmail.com -src fiverr.com -dd 30/02/2075`
```
Comment (immanuelhume) [inline docs/UserGuide.md:486]:
```suggestion
**Create a project** | `p -pn project_name -e client_email [-s status] [-src source] [-d description] [-ad accepted_date] [-dd deadline_date]` | `p -pn Mycelium Desktop -e spiderman@gmail.com -src fiverr.com -dd 30/02/2075`
```

File: docs/UserGuide.md [hunk 4]
```diff
@@ -464,9 +478,12 @@ If you have a project or client selected, pressing ***Enter*** in **search mode*
...
+**Delete a project** | dp -pn project_name | `dp -pn Mycelium Desktop`
```
Comment (immanuelhume) [inline docs/UserGuide.md:487]:
```suggestion
**Delete a project** | `dp -pn project_name` | `dp -pn Mycelium Desktop`
```

File: docs/UserGuide.md [hunk 5]
```diff
@@ -464,9 +478,12 @@ If you have a project or client selected, pressing ***Enter*** in **search mode*
...
+**Update an existing project** | up -pn project_name [-e email] [-s status] [-src source] [-d description]  [-ad accepted_date] [-dd deadline_date] [-pn2 new_project_name] | `up -pn Mycelium Desktop -e alice_baker@bakers.com -dd 30/03/2024`
```
Comment (immanuelhume) [inline docs/UserGuide.md:488]:
```suggestion
**Update an existing project** | `up -pn project_name [-e email] [-s status] [-src source] [-d description]  [-ad accepted_date] [-dd deadline_date] [-pn2 new_project_name]` | `up -pn Mycelium Desktop -e alice_baker@bakers.com -dd 30/03/2024`
```

File: docs/UserGuide.md [hunk 6]
```diff
@@ -464,9 +478,12 @@ If you have a project or client selected, pressing ***Enter*** in **search mode*
...
+**Fuzzy Search** | (CTRL + F) query | Press (CTRL + F) then enter Alice
```
Comment (immanuelhume) [inline docs/UserGuide.md:489]: i'm thinking it's probably better to leave this out from the command summary table. if we want, we can maybe have a hotkey summary table for the hotkeys

File: docs/UserGuide.md [hunk 7]
```diff
@@ -464,9 +478,19 @@ If you have a project or client selected, pressing ***Enter*** in **search mode*
...
+* `q` is the command,
+* `-a foo` indicates a compulsory argument,
+* `[-b bar]` indicates an optional argument.
+
+
+Action | Format | Example 
+--------|------------------|------
+[**Create new client**](#creating-a-client-contact--c) | `c -cn client_name -e email [-y year_of_birth] [-src source] [-mn mobile_number]` | `c -cn Alice Baker -e alice_baker@bakers.com -y 2000`
+[**Delete an existing client** ](#deleting-a-client-contact--dc)| dc -e email | `dc -e alice_baker@bakers.com`
+**Update an existing client** | uc -e email [-cn client_name] [-y year_of_birth] [-src source] [-mn mobile_number] [-e2 new_email] | `uc -e alice_baker@bakers.com -y 2002 -mn 85090234`
```
Comment (StanleyNeoh) [inline docs/UserGuide.md:493]: will need to add link once update client part of UG is done

**Review by StanleyNeoh — APPROVED** [review]

**Pull request merged**